pkg/metrics/github_fetcher.go

```diff
@@ -102,39 +102,38 @@ func periodicGithubFetcher() {
 
 		// Fetch repositories (if dynamic)
 		var repos_to_fetch []string
-		var current_repos_per_org = make(map[string]int)
+		var current_repos_per_org = make(map[string][]string)
+		var r []string
 
 		if len(config.Github.Repositories.Value()) > 0 {
 			repos_to_fetch = config.Github.Repositories.Value()
 		} else {
 			for _, orga := range config.Github.Organizations.Value() {
-				c, exist := repos_per_org[orga]
+				repos, exist := repos_per_org[orga]
+				previousCount := len(repos)
 				currentCount := countAllReposForOrg(orga)
-				if !exist || c != currentCount {
-					repos_to_fetch = append(repos_to_fetch, getAllReposForOrg(orga)...)
+				if !exist || previousCount != currentCount {
+					log.Printf("getAllReposForOrg count updated from %d to %d", previousCount, currentCount)
+					r = getAllReposForOrg(orga)
 				} else {
-					log.Printf("Skipping getAllReposForOrg, repo count unchanged %d", c)
+					log.Printf("Skipping getAllReposForOrg, repo count unchanged: %d", previousCount)
+					r = repos_per_org[orga]
 				}
-				current_repos_per_org[orga] = currentCount
+				repos_to_fetch = append(repos_to_fetch, r...)
+				current_repos_per_org[orga] = r
 			}
 		}
 		repositories = repos_to_fetch
 		repos_per_org = current_repos_per_org
 
 		// Fetch workflows
-		non_empty_repos := make([]string, 0)
 		ww := make(map[string]map[int64]github.Workflow)
 		for _, repo := range repos_to_fetch {
 			r := strings.Split(repo, "/")
 			workflows_for_repo := getAllWorkflowsForRepo(r[0], r[1])
-			if len(workflows_for_repo) == 0 {
-				continue
-			}
-			non_empty_repos = append(non_empty_repos, repo)
 			ww[repo] = workflows_for_repo
 			log.Printf("Fetched %d workflows for repository %s", len(ww[repo]), repo)
 		}
-		repositories = non_empty_repos
```

Comment:
i think we should revert this, b/c it will keep `repositories` as expected: to be full of non workflow empty repos and then have the count short-circuit as an add-on w/ new variables

Reply:
https://github.com/search?q=repo%3Afaubion-hbo%2Fgithub-actions-exporter+%2Frepositories%2F&type=code shows it's used elsewhere, so keep those operating the same

Reply:
agree